python/cugraph/tests/test_hits.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+import time
+import numpy as np
+import pandas as pd
+
+import pytest
+
+import cudf
+import cugraph
+from cugraph.tests import utils
+
+# Temporarily suppress warnings till networkX fixes deprecation warnings
+# (Using or importing the ABCs from 'collections' instead of from
+# 'collections.abc' is deprecated, and in 3.8 it will stop working) for
+# python 3.7.  Also, this import networkx needs to be relocated in the
+# third-party group once this gets fixed.
+import warnings
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import networkx as nx
+
+
+print('Networkx version : {} '.format(nx.__version__))
+
+
+def cudify(d):
+    if d is None:
+        return None
+
+    k = np.fromiter(d.keys(), dtype='int32')
+    v = np.fromiter(d.values(), dtype='float32')
+    cuD = cudf.DataFrame({'vertex': k, 'values': v})
+    return cuD
+
+
+def cugraph_call(cu_M, max_iter, tol):
+    # cugraph hits Call
+    G = cugraph.DiGraph()
+    G.from_cudf_edgelist(cu_M, source='0', destination='1')
+    t1 = time.time()
+    df = cugraph.hits(G, max_iter, tol)
+    t2 = time.time() - t1
+    print('Cugraph Time : '+str(t2))
+
+    return df
+
+
+# The function selects personalization_perc% of accessible vertices in graph M
+# and randomly assigns them personalization values
+def networkx_call(M, max_iter, tol):
+    # in NVGRAPH tests we read as CSR and feed as CSC,
+    # so here we do this explicitly
+    print('Format conversion ... ')
+
+    # Directed NetworkX graph
+    Gnx = nx.from_pandas_edgelist(M, source='0', target='1',
+                                  create_using=nx.DiGraph())
+
+    # Networkx Hits Call
+    print('Solving... ')
+    t1 = time.time()
+
+    # same parameters as in NVGRAPH
+    pr = nx.hits(Gnx, max_iter, tol, normalized=True)
+    t2 = time.time() - t1
+
+    print('Networkx Time : ' + str(t2))
+
+    return pr
+
+
```

Comment:
I'm trying to get all tests to use a common set of data files.   Please use   utils.DATASETS

Reply:
Two of the data sets in the test_utils.DATASETS list fail this test.

1. polbooks.csv raises an exception (power iteration doesn't converge).  Since gunrock hits doesn't attempt to converge, just runs a fixed number of iterations... it generates an answer.
1. netscience creates a different answer.  Would need to investigate why the answers are different.  It's not subtle, some of the answers in netscience are null, some of the vertices have scores that are large where the gunrock score is zero.

Not sure how we can accommodate issue 1 when all tests use the same data.  It's certainly possible that a data set might not converge in a power iteration algorithm.

I will need to investigate further why the netscience data set creates a totally different answer.